### PR TITLE
feat(backend): Adding a cron job for MariaDB backup

### PR DESCRIPTION
**Pull request by DharmitD**
**Which issue is resolved by this Pull Request:** 
Resolves # [RHODS-6117](https://issues.redhat.com/browse/RHODS-6117)

**Description of your changes:**

- Implemented a Cron Job that creates a dump file for disaster recovery / back up of the MariaDB database. Dump file is stored in the Cron Job pod.
- Updated ServiceAccounts, Roles and RoleBindings such that the CronJob has permissions to create and store the dump file.

**Checklist:**
- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  labels:
+    application-crd-id: data-science-pipelines
+  name: mariadb-cron
```
Comment (HumairAK) [inline manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: seems like a non-meaningful name, we know it's a cronjob based on it's kind, I would name it something like "mariadb-backup"

File: manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: mariadbdump
+rules:
+  - apiGroups:
+      - ""
+      - ''
```
Comment (HumairAK) [inline manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml]: only need one of these entries

File: manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    resources:
+      - deployments
+      - replicasets
```
Comment (HumairAK) [inline manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml]: `deployments` and `replicasets` are not in the core (denoted by `""`) group, they are in `apps`

File: manifests/opendatahub/overlays/metadata-store-mariadb/rolebindings/mariadb-binding.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: mariadbdump
+subjects:
+  - kind: ServiceAccount
+    name: mariadbdumpsa
+roleRef:
+  kind: Role
+  name: mariadbdump
+  apiGroup: ""
```
Comment (HumairAK) [inline manifests/opendatahub/overlays/metadata-store-mariadb/rolebindings/mariadb-binding.yaml]: the apiGroup should be `rbac.authorization.k8s.io`

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                - name: MYSQL_ALLOW_EMPTY_PASSWORD
+                  value: "true"
+              volumeMounts:
+                - name: mariadb-persistent-storage
+                  mountPath: /var/lib/mysql
+          serviceAccount: mariadbdumpsa
+          volumes:
+            - name: mariadb-persistent-storage
+              persistentVolumeClaim:
+                claimName: mariadb-pv-claim
```
Comment (HumairAK) [inline manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: why are we mounting the mariadb pv here? aren't we trying to backup the data in this pv to another location? 

also this will fail as soon as the cronjob pod is scheduled on a node that is different from mariadb due to the pv being `RWO` access mode.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+            parent: "mariadb-cron"
+        spec:
+          containers:
+            - name: mariadb
+              image: mysql
+              command:
+                - /bin/sh
+                - '-c'
+                - >
+                  mysqldump -h mysql -u $MYSQL_USER --password=$MYSQL_PASSWORD
```
Comment (HumairAK) [inline manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: >  mysqldump -h mysql -u $MYSQL_USER --password=$MYSQL_PASSWORD --all-databases > mariadb-backup.sql; sleep 6000

doesn't look like this is going into the pv you are mounting (is that what you wanted?), it just looks like it's redirected to a file in the working directory

File: manifests/opendatahub/overlays/metadata-store-mariadb/serviceaccounts/mariadb.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: mariadbdumpsa
```
Comment (gmfrasca) [inline manifests/opendatahub/overlays/metadata-store-mariadb/serviceaccounts/mariadb.yaml]: We should either leverage the pre-existing `mysql` SA, which doesn't appear to have an actual Role/RoleBinding attached, or remove the `mysql` SA and rename this one to just `mariadb`

File: manifests/opendatahub/overlays/metadata-store-mariadb/rolebindings/mariadb-binding.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
+kind: RoleBinding
```
Comment (gmfrasca) [inline manifests/opendatahub/overlays/metadata-store-mariadb/rolebindings/mariadb-binding.yaml]: Is this RB actually needed since the CJ accesses mariadb via service now?

File: manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
+kind: Role
```
Comment (gmfrasca) [inline manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml]: Is this Role actually needed since the CJ accesses mariadb via service now?

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+                  --all-databases > mariadb-backup.sql; sleep 6000
+              env:
+                - name: MYSQL_USER
+                  valueFrom:
+                    secretKeyRef:
+                      name: mysql-secret
```
Comment (gmfrasca) [inline manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: this should use the variable reference `$(database_secret)` instead of the hardcoded `mysql-secret`

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+                      key: username
+                - name: MYSQL_PASSWORD
+                  valueFrom:
+                    secretKeyRef:
+                      name: mysql-secret
```
Comment (gmfrasca) [inline manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: this should use the variable reference `$(database_secret)` instead of the hardcoded `mysql-secret`

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+                      key: password
+                - name: MYSQL_DATABASE
+                  valueFrom:
+                    configMapKeyRef:
+                      key: pipelineDb
+                      name: pipeline-install-config
```
Comment (gmfrasca) [inline manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: this should use the variable reference `$(pipeline_install_configuration)` instead of the hardcoded `mysql-secret`

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+  failedJobsHistoryLimit: 1
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            parent: "mariadb-cron"
+        spec:
+          containers:
+            - name: mariadb
```
Comment (gmfrasca) [inline manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: should we name this something like `mariadb-backup` so we know it's not a vanilla mariadb instance?

File: manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml (hunk 3 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml]: would I also need to add `batch` along with `apps` in the `apiGroups`, since the cronjob's apiVersion is `batch/v1`?

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml (hunk 1 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: Done, updated the name to be "mariadb-backup"

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml (hunk 5 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: Taking the volumes/volume mounts/pvs out as per our discussion.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml (hunk 6 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: Taking the volumes/volume mounts/pvs out as per our discussion.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml (hunk 11 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: Done, updated to have the variable reference instead of the hardcoded secret value.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml (hunk 12 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: Done, updated to have the variable reference instead of the hardcoded secret value.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml (hunk 10 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: Done, updated to have the variable reference instead of the hardcoded secret value.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml (hunk 13 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-cron.yaml]: Done, renamed to `mariadb-backup`.

File: manifests/opendatahub/overlays/metadata-store-mariadb/serviceaccounts/mariadb.yaml (hunk 7 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/serviceaccounts/mariadb.yaml]: Tested out with `mysql` SA on my cluster and everything worked as expected - cron job pod came up and created the dump file without and permission issues. Hence removing this SA, and the associated role/role binding, using the pre-existing `mysql` SA instead.

File: manifests/opendatahub/overlays/metadata-store-mariadb/rolebindings/mariadb-binding.yaml (hunk 8 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/rolebindings/mariadb-binding.yaml]: Tested out with mysql SA on my cluster and everything worked as expected - cron job pod came up and created the dump file without and permission issues. Hence removing this SA, and the associated role/role binding, using the pre-existing mysql SA instead.

File: manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml (hunk 9 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml]: Tested out with mysql SA on my cluster and everything worked as expected - cron job pod came up and created the dump file without and permission issues. Hence removing this SA, and the associated role/role binding, using the pre-existing mysql SA instead.

File: manifests/opendatahub/overlays/metadata-store-mariadb/rolebindings/mariadb-binding.yaml (hunk 4 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/rolebindings/mariadb-binding.yaml]: Taking out the role binding based on discussion [here](https://github.com/opendatahub-io/data-science-pipelines/pull/61#discussion_r1049997819)

File: manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml (hunk 3 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml]: Taking out the role based on discussion in https://github.com/opendatahub-io/data-science-pipelines/pull/61#discussion_r1049997819

File: manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml (hunk 2 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/roles/mariadb-role.yaml]: Taking out the role based on discussion in https://github.com/opendatahub-io/data-science-pipelines/pull/61#discussion_r1049997819

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+            parent: "mariadb-backup"
+        spec:
+          containers:
+            - name: mariadb-backup
+              image: mysql
+              command:
+                - /bin/sh
+                - '-c'
+                - >
+                  mysqldump -h mysql -u $MYSQL_USER --password=$MYSQL_PASSWORD
```
Comment (accorvin) [inline manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml:28]: IMO we shouldn't commit the sleep here into master. That means every backup run will take a long time. We'll have to figure out a way to make this work with the tests reliably

File: tests/basictests/ds-pipelines.sh [hunk 15]
```diff
@@ -12,12 +12,22 @@ os::test::junit::declare_suite_start "$MY_SCRIPT"
...
     os::cmd::expect_success "if [ "$running_pods" -gt "0" ]; then exit 0; else exit 1; fi"
 }
 
+function check_mariadb_backup() {
+    header "Check if MariaDB backup comes up correctly"
+    os::cmd::expect_success "oc get cronjob mariadb-backup"
+    oc create job --from=cronjob/mariadb-backup mariadb-backup
+    os::cmd::try_until_text "oc get pods | grep Running | grep 'mariadb-backup' | awk '{print \$1}' | wc -w" "1" $odhdefaulttimeout $odhdefaultinterval
+    oc rsync $(oc get pods | grep Running | grep 'mariadb-backup' | awk '{print $1}'):mariadb-backup.sql .
+    os::cmd::expect_success "if [ -f "mariadb-backup.sql" ]; then exit 0; else exit 1; fi"
```
Comment (accorvin) [inline tests/basictests/ds-pipelines.sh]: It would also be good to add a test to ensure that the backup file isn't empty. Maybe check the size?

Comment (HumairAK) [inline tests/basictests/ds-pipelines.sh]: you don't really need to rsync this backup to test if it exists, you can just do: 

`oc exec -ti $backup_pod -- /bin/sh -c "if [ -f "mariadb-backup.sql" ]; then exit 0; else exit 1; fi"`

in your method, you will always need to ensure test pod has enough space in its fs to store the backup, this way you don't need to worry about that.

Comment (HumairAK) [reply tests/basictests/ds-pipelines.sh]: seems like you also run  `$(oc get pods | grep Running | grep 'mariadb-backup' | awk '{print $1}')` twice, you can instead execute it once and store it as a variable

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+                      name: $(database_secret)
+                      key: password
+                - name: MYSQL_DATABASE
+                  valueFrom:
+                    configMapKeyRef:
+                      key: pipelineDb
+                      name: $(pipeline_install_configuration)
+                - name: MYSQL_ALLOW_EMPTY_PASSWORD
+                  value: "true"
+          serviceAccount: mysql
```
Comment (HumairAK) [inline manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml]: 1) serviceaccount is deprecated, use `ServiceAccountName` instead
2) do you need `mysql` serviceaccount for this? seems like you don't need any additional permission beyond what default SA has, have you tried this job w/o specifying an SA?

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml (hunk 14 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml:28]: I'd added the sleep because the pods would move into "complete" state as soon as they Cron Job spun them up. But that was maybe because I'd scheduled the Cron Job to run every 5 minutes when scripting/debugging it. Trying without the sleep statement now, I think it should work with the Cron Job set to run once daily.

Comment (HumairAK) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml:28]: Sounds like the backup job was just completing fast, and that `sleep` actually doesn't really do anything here.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml (hunk 16 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml]: Tested out without the SA, Cron Job comes up correctly and works as expected. Taking the SA out.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml (hunk 14 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml:28]: Correct, `sleep` was just preventing the job from completing fast when scheduled for every 5 minutes. Tested out and confirmed that the CJ works without the sleep in our case, i.e., when scheduled to run once daily.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml (hunk 16 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml]: Okay, I take that back. I tested out again and without the SA, I see this permissions denied error:
`/bin/sh: mariadb-backup.sql: Permission denied`

I've seen this error before and that originally led me to adding an SA.

You can also find the tests failing because they aren't able to find a running Cron Job pod.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml (hunk 14 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml:28]: `sleep` prevents the cron job prod to not get into a `completed` state quickly (a couple of seconds), else CJ just runs the mysqldump command and rushes to move pod to a `completed` state.

We can't exec /rsh into completed pods, hence the tests would fail without the `sleep`. We would need to think of a workaround to avoid the cron job pod completing fast.

Comment (accorvin) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml:28]: A potential solution that jumps out to me: the command can consume some environment variable that determines how long/whether to sleep. In production it would be set to not sleep, or not sleep for very long. In testing, it could be set to sleep for a sufficiently long time as to ensure that the tests can rsh and check the file.

This is a quick and dirty solution that would work, but talk with the team to see whether you all can think of something more elegant.

File: tests/basictests/ds-pipelines.sh (hunk 15 above)
Comment (DharmitD) [reply tests/basictests/ds-pipelines.sh]: Done, updated the if condition to check that the file exists and is not empty, using the "-s" option [1].

[1] https://www.cyberciti.biz/faq/linux-unix-script-check-if-file-empty-or-not/

Comment (DharmitD) [reply tests/basictests/ds-pipelines.sh]: Done, updated to take the rsync out and have `oc exec` instead. Also stored the command in a variable, but I also had to add a small (30s) sleep because it seems the variable initialization happens quickly but the Cron Job pod doesn't come up until then, so it fails without the sleep.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml (hunk 16 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml]: Tested out with `serviceAccountName` instead and that worked, was able to access the backup file.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml (hunk 14 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml:28]: This seems to be a good idea, could set a flag this way to have sleep only for tests.

Although, even with the 6000s sleep cycle added, it still takes very less time for the Cron Job to create the backup file. Here's logs from a local test run I tried (I ideally wanted to point to OpenShift CI's test artifacts but looks like their server's down):

```
Check if MariaDB backup comes up correctly

Running operator-tests/data-science-pipelines/basictests/ds-pipelines.sh:23: executing 'oc get cronjob mariadb-backup' expecting success...


✔ SUCCESS after 1.086s: operator-tests/data-science-pipelines/basictests/ds-pipelines.sh:23: executing 'oc get cronjob mariadb-backup' expecting success

job.batch/mariadb-backup-2 created
Running operator-tests/data-science-pipelines/basictests/ds-pipelines.sh:27: executing 'oc exec -ti mariadb-backup-2-lc754 -- /bin/sh -c "if [ -s mariadb-backup.sql ]; then exit 0; else exit 1; fi"' expecting success...


✔ SUCCESS after 3.241s: operator-tests/data-science-pipelines/basictests/ds-pipelines.sh:27: executing 'oc exec -ti mariadb-backup-2-lc754 -- /bin/sh -c "if [ -s mariadb-backup.sql ]; then exit 0; else exit 1; fi"' expecting success
```
It takes ~30s for the Cron Job pod to come up and then the backup file seems to be instantly generated.

This behavior would stay the same even with a PV created to store the backup file I'm assuming, backup file would get created and the CJ pod would still keep running for 6000s (or how much ever time we set the sleep cycle to run for, ideally reduce it to time just enough for tests to find the back up file in the CJ pod). I'm just trying to think of why the backup process might take longer with the sleep cycle added in Prod. Am I missing something? Or is it considered a bad practice in general to add a sleep cycle?

Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml:28]: Looks like the server's up, you could find the logs [here.](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/opendatahub-io_data-science-pipelines/61/pull-ci-opendatahub-io-data-science-pipelines-master-data-science-pipelines-e2e/1613233874656038912/artifacts/data-science-pipelines-e2e/e2e/build-log.txt)

Comment (accorvin) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml:28]: @DharmitD to your question of why we should avoid adding the sleep cycle in production: yes it's primarily just considered a bad practice. It means we'll needlessly have pods running on a cluster for longer than they need to be, wasting resources.

File: tests/basictests/ds-pipelines.sh (hunk 15 above)
Comment (HumairAK) [reply tests/basictests/ds-pipelines.sh]: avoid sleep, for reasons alex mentioned above amongst others, we can't control for `sleep` for many reasons and it should always be a last resort, I think you can use the library functions such as `os::cmd::expect_success` to ensure the pod has come up before executing your commands

Comment (DharmitD) [reply tests/basictests/ds-pipelines.sh]: Done, took the sleep out, as well as the `oc exec` into the CJ pod, and just kept checks around the CJ resource, as well as the job and pod created out of it, as per our discussion.

File: manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml (hunk 14 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mariadb/cronjobs/mariadb-backup.yaml:28]: Makes sense, took the sleep out, will add checks around the backup file after creating a storage / PV for the backup, as per our discussion.

File: tests/basictests/ds-pipelines.sh (hunk 15 above)
Comment (DharmitD) [reply tests/basictests/ds-pipelines.sh]: Will add checks around the backup file after creating a storage / PV for the backup, as per our discussion.

**Review by gmfrasca — COMMENTED** [review]
/approve
